### ISSUE TITLE
Box3d evaluation error 

### ISSUE THREAD

**Issue by DeriZSY** [traveller59/second.pytorch#90]
I get strange results when using the evaluation tools to evaluate our own kitti-format data. I input exactly the same annotations for gt_annos and dt_annos, but got wrong result for bev and 3d. Here's what I got: 

Car AP@0.70, 0.70, 0.70:
bbox AP:100.00, 100.00, 100.00
bev  AP:1.49, 1.45, 1.45
3d   AP:0.63, 0.60, 0.60
Car AP@0.70, 0.50, 0.50:
bbox AP:100.00, 100.00, 100.00
bev  AP:2.27, 2.22, 2.22
3d   AP:0.63, 0.60, 0.60

Could it be a problem that I input detection results in world frame (instead of camera frame)? 

I check the source code and really cannot understand why we need "rotate iou evaluate" instead of directly compute the intersection parts of 3d boxes. 

Hope to get some suggestions. 
Thanks very much!

**Comment by traveller59** [comment]
I think this may due to rotate iou algorithm, it can't handle this kind of fake data, but it should works with actual data.

the "rotate iou evaluate" add option for calculate intersection instead of return iou, which is used for 3D iou calculation.

**Comment by DeriZSY** [comment]
> I think this may due to rotate iou algorithm, it can't handle this kind of fake data, but it should works with actual data.
> 
> the "rotate iou evaluate" add option for calculate intersection instead of return iou, which is used for 3D iou calculation.

Thanks for your reply. I've confirmed that the problem is due to rotate iou. The smaller the rotation angle, the higher the accuracy (when input gt for both gt_annos and dt_annos).

Do you have any idea what should I do with the rotation angle to fix it? Should I transfer it to some other data format?

**Comment by DeriZSY** [comment]
I tried to set the position, rotation_y, dimension to exactly the same value for all detections. And found even back to `rotate_iou_gpu_eval()`, I got everything fine. 
i.e.   `print(boxes.any() == query_boxes.any()) ` yields "True" all the time. 

But as the value passed to `rotate_iou_kernel_eval()`,  I begin to got large values (ranging from 4 to -400) for 
`           dx = block_qboxes[i * 5] - block_boxes[i * 5]
            dz = block_qboxes[i * 5+1] - block_boxes[i * 5+1]
            dl = block_qboxes[i * 5+2] - block_boxes[i * 5+2]
            dw = block_qboxes[i * 5+3] - block_boxes[i * 5+3]
            dr = block_qboxes[i * 5+4] - block_boxes[i * 5+4] 
            print(dx+dz+dl+dw+dr)
`
Could it be the problem of value passing on GPU?
